client/components/Signup.jsx

```diff
@@ -1,9 +1,88 @@
+import { useState } from 'react';
+import { useNavigate } from "react-router-dom";
+import axios from 'axios';
+import styles from './Signup.module.css';
 
+const Signup = () => {
 
-function Signup() {
-  return (
-    <div>This is the Signup page</div>
-  )
-}
+    const [inputUserName, setInputUserName] = useState('');
+    const [inputEmail, setInputEmail] = useState('');
+    const [inputPassword, setInputPassword] = useState('');
+    const [invalidUsername, setInvalidUsername] = useState(false);
+    const [invalidEmail, setInvalidEmail] = useState(false);
+    const [invalidPassword, setInvalidPassword] = useState(false);
 
-export default Signup
\ No newline at end of file
+    const navigate = useNavigate();
+
+    const onSubmitHandler = (event) => {
+
+        event.preventDefault();
+        postResponse();
+    };
+
+    const postResponse = async () => {
+
+        const formData = { userName: inputUserName, email: inputEmail, password: inputPassword }
+
+        if (validateEmail(inputEmail) && validatePassword(inputPassword) && validateUsername(inputUserName)) {
+            try {
+                const res = await axios.post('/api/signup', formData);
+                console.log(res);
+            } catch (err) { console.log(err); }
+            navigate("/");
```

Comment:
does this navigates to the homepage ?
maybe it should be in the 'try' block ?

correct me if im wrong, if user registers correctly, he goes to homepage.
when he gets error , he still goes to homepage ?

Reply:
You are completely right, good find!